PerimeterXModule/Internals/PxContext.cs

```diff
@@ -7,167 +7,257 @@
+		public string Hostname { get; set; }
+		public string Uri { get; set; }
+		public string UserAgent { get; set; }
+		public string FullUrl { get; set; }
+		public string S2SCallReason { get; set; }
+		public int Score { get; set; }
+		public string Vid { get; set; }
+		public string UUID { get; set; }
+		public BlockReasonEnum BlockReason { get; set; }
+		public bool MadeS2SCallReason { get; set; }
+		public string S2SHttpErrorMessage { get; set; }
+		public string BlockAction { get; set; }
+		public string BlockData { get; set; }
+		public HttpContext ApplicationContext { get; private set; }
+		public bool SensitiveRoute { get; set; }
+		public PassReasonEnum PassReason { get; set; }
+		public long RiskRoundtripTime { get; set; }
+		public CookieOrigin CookieOrigin { get; set; }
+		public Dictionary<string,string> OriginalTokens { get; set; }
+		public IPxCookie OriginalToken { get; set; }
+		public string OriginalTokenError { get; set; }
+		public string OriginalUUID { get; set; }
+		public object DecodedOriginalToken { get; set; }
+
+		public PxContext(HttpContext context, PxModuleConfigurationSection pxConfiguration)
+		{
+			ApplicationContext = context;
+
+			CookieOrigin = CookieOrigin.COOKIE;
+			PxCookies = new Dictionary<string, string>();
+			OriginalTokens = new Dictionary<string, string>();
+			S2SCallReason = "none";
+
+			// Get Headers
+
+			// if userAgentOverride is present override the default user-agent
+			string userAgentOverride = pxConfiguration.UserAgentOverride;
+			if (!string.IsNullOrEmpty(userAgentOverride))
+			{
+				UserAgent = context.Request.Headers[userAgentOverride];
+			}
+			// ua fallback
+			if (string.IsNullOrEmpty(UserAgent))
+			{
+				UserAgent = context.Request.Headers["user-agent"];
+			}
+
+			Headers = new List<RiskRequestHeader>();
+
+			foreach (string header in context.Request.Headers.Keys)
+			{
+				if (!pxConfiguration.SensitiveHeaders.Contains(header))
+				{
+					RiskRequestHeader riskHeader = new RiskRequestHeader
+					{
+						Name = header,
+						Value = context.Request.Headers.Get(header)
+					};
+					if (header.ToLower() == "user-agent")
+					{
+						riskHeader.Value = UserAgent;
+					}
+					Headers.Add(riskHeader);
+				}
+			}
+
+			// Handle Cookies
+			var contextCookie = context.Request.Cookies;
+			string mobileHeader = context.Request.Headers[PxConstants.MOBILE_HEADER];
+			// Check if X-PX-AUTHORIZATION exist
+			if (mobileHeader != null)
+			{
+				// Extract Original Tokens
+				CookieOrigin = CookieOrigin.HEADER;
+				string originalToken = context.Request.Headers[PxConstants.ORIGINAL_TOKEN];
+				if (!string.IsNullOrEmpty(originalToken))
+				{
+					string[] splitedOriginalToken = originalToken.Split(MOBILE_DELIMITER, 2);
+					string[] fallbackSplittedOriginalToken = originalToken.Split(MOBILE_DELIMITER);
+					if (!string.IsNullOrEmpty(splitedOriginalToken[0]) && Array.IndexOf(PxConstants.PX_TOKEN_PREFIX, splitedOriginalToken[0]) > -1)
```

Comment:
if you've passed the upper if - `!string.IsNullOrEmpty(originalToken) == true` then originalToken is not empty. As such, doing a `Split` on it will necessarily return an array of at least one value (see [here](https://msdn.microsoft.com/en-us/library/c1bs0eda(v=vs.110).aspx), "Banana" example). 

Bottom line - checking `!string.IsNullOrEmpty(splitedOriginalToken[0])` is not required.